New cluster property - hazelcast.journal.cleanup.threshold [CTT-761, CTT-739] (#1946)
hazelcast/hz-docs: docs/modules/ROOT/pages/system-properties.adoc

|int
|Some JMX MBeans are cached  to reduce the overhead of calculating statistics. This parameter determines for how long the MBeans can go stale.

|`hazelcast.journal.cleanup.threshold`
| 1.0
| float
| The cleanup threshold for the event journal, expressed as a fraction of its capacity (a value between 0.0 and 1.0).

> When the remaining capacity falls below this threshold, a cleanup may be triggered during put operations to remove expired events (only if a xref:data-structures:event-journal.adoc#configuring-event-journal-ttl[time-to-live] for event journal is configured).

Reviewer: > during put operations

I am wondering (maybe nit-picking) if it is clear that we mean here journal/ringbuffer put operation and not IMap/ICache put operation

|int
|Some JMX MBeans are cached  to reduce the overhead of calculating statistics. This parameter determines for how long the MBeans can go stale.

|`hazelcast.journal.cleanup.threshold`

> | 1.0f

Reviewer: the `f` suffix is not necessary when configuring the property and may be puzzling for for non-java users

| float

> | The cleanup threshold for the event journal, expressed as a fraction of its capacity (a value between 0.0f and 1.0f).

Reviewer: ```suggestion
| The cleanup threshold for the event journal, expressed as a fraction of its capacity (a value between 0.0 and 1.0).
```

> When the remaining capacity falls below this threshold, a cleanup may be triggered during put operations to remove expired events (only if a TTL is configured).

Reviewer: ```suggestion
When the remaining capacity falls below this threshold, a cleanup may be triggered during put operations to remove expired events (only if a `timeToLiveSeconds` for event journal is configured).
```

Reviewer: maybe add link to the [[configuring-event-journal-ttl]] section?

> A value of 0.0f means cleanup occurs when the journal is full, while a value of 1.0f means cleanup is triggered after each put operation.

Reviewer: > A value of 0.0f means cleanup occurs when the journal is full

It rather means that there is no cleanup, only overwriting the entries when the journal fills

Author: d6a3e1ff830590d327569846a6d1ddfb31963fd0

> A value of 0.0f means cleanup occurs when the journal is full, while a value of 1.0f means cleanup is triggered after each put operation.

Reviewer: ```suggestion
A value of 0.0 means cleanup occurs when the journal is full, while a value of 1.0 means cleanup is triggered after each put operation.
```

> Applies to event journals of both IMap and ICache.

Reviewer: ```suggestion
```

It applies to all data structures having journal (currently IMap and ICache) and the design is generic and with the idea that other structures would behave in the same way. So I would remove this explicit enumeration.
